crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -64,6 +66,25 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
             .map(ForeignInto::foreign_into)
             .or(payment_intent.setup_future_usage);
 
+        if auth_flow == services::AuthFlow::Client && request.customer_id.is_some() {
+            let response = request
+                .clone()
+                .customer_id
+                .and_then(|customer| {
+                    payment_intent
+                        .clone()
+                        .customer_id
+                        .map(|payment_customer| payment_customer != customer)
+                })
+                .unwrap_or_default();
+            if response {
+                return Err(errors::ApiErrorResponse::GenericUnauthorized {
+                    message: "Unauthorised access to update customer".to_string(),
+                }
+                .into());
+            }
+        }
```

Comment:
this logic could have been moved to a helper function in order to avoid duplication, or else apply the same set of changes as suggested above.

Reply:
done!